document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutorTest.java

```diff
@@ -0,0 +1,306 @@
+import org.hypertrace.core.documentstore.query.Pagination;
+import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.documentstore.query.Selection;
+import org.hypertrace.core.documentstore.query.SelectionSpec;
+import org.hypertrace.core.documentstore.query.Sort;
+import org.hypertrace.core.documentstore.query.SortingSpec;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers
+public class MongoQueryExecutorTest {
+  private static final String COLLECTION_NAME = "mongoQueryExecutorTest";
+
+  private static GenericContainer<?> mongo;
+  private static Collection collection;
+
+  @BeforeAll
+  public static void init() throws IOException {
+    mongo =
+        new GenericContainer<>(DockerImageName.parse("mongo:4.4.0"))
+            .withExposedPorts(27017)
+            .waitingFor(Wait.forListeningPort());
+    mongo.start();
+
+    DatastoreProvider.register("MONGO", MongoDatastore.class);
+
+    Map<String, String> mongoConfig = new HashMap<>();
+    mongoConfig.putIfAbsent("host", "localhost");
+    mongoConfig.putIfAbsent("port", mongo.getMappedPort(27017).toString());
+    Config config = ConfigFactory.parseMap(mongoConfig);
+
+    Datastore datastore = DatastoreProvider.getDatastore("Mongo", config);
+
+    datastore.deleteCollection(COLLECTION_NAME);
+    datastore.createCollection(COLLECTION_NAME, null);
+    collection = datastore.getCollection(COLLECTION_NAME);
+
+    Map<Key, Document> documents = createDocumentsFromResource("mongo/collection_data.json");
+    collection.bulkUpsert(documents);
+  }
+
+  @AfterAll
+  public static void shutdown() {
+    mongo.stop();
+  }
+
+  @Test
+  public void testFindAll() throws IOException {
+    Query query = Query.builder().build();
+
+    Iterator<Document> resultDocs = collection.find(query);
+    assertSizeEqual(resultDocs, "mongo/collection_data.json");
+  }
+
+  @Test
+  public void testFindSimple() throws IOException {
+    List<SelectionSpec> selectionSpecs =
+        List.of(
+            SelectionSpec.of(IdentifierExpression.of("item")),
+            SelectionSpec.of(IdentifierExpression.of("price")),
+            SelectionSpec.of(IdentifierExpression.of("quantity")),
+            SelectionSpec.of(IdentifierExpression.of("date")));
+    Selection selection = Selection.builder().selectionSpecs(selectionSpecs).build();
+    Filter filter =
+        Filter.builder()
+            .expression(
+                RelationalExpression.of(
+                    IdentifierExpression.of("item"),
+                    NOT_IN,
+                    ConstantExpression.ofStrings(List.of("Soap", "Bottle"))))
```

Comment:
nit: for completeness that it matches more than one, we could have one more matching `shampoo in the list.

Reply:
Just saw that we have a test for this later.